integrationtests/src/test/java/io/kroxylicious/proxy/schema/validation/JsonSyntaxValidationIT.java

```diff
@@ -0,0 +1,301 @@
+        String proxyAddress = "localhost:9192";
+
+        admin.createTopics(List.of(new NewTopic(TOPIC_1, 1, (short) 1), new NewTopic(TOPIC_2, 1, (short) 1))).all().get();
+
+        var config = baseConfigBuilder(proxyAddress, cluster.getBootstrapServers()).addNewFilter().withType("ProduceValidator").withConfig(Map.of("rules",
+                List.of(Map.of("topicNames", List.of(TOPIC_1), "valueRule",
+                        Map.of("allowsNulls", true, "syntacticallyCorrectJson", Map.of("validateObjectKeysUnique", true))))))
+                .endFilter()
+                .build().toYaml();
+
+        try (var ignored = startProxy(config)) {
+            try (var producer = getProducer(proxyAddress, 0)) {
+                Future<RecordMetadata> invalid = producer.send(new ProducerRecord<>(TOPIC_1, "my-key", SYNTACTICALLY_INCORRECT_JSON));
+                assertInvalidRecordExceptionThrown(invalid, "value was not syntactically correct JSON");
+
+                producer.send(new ProducerRecord<>(TOPIC_2, "my-key", SYNTACTICALLY_INCORRECT_JSON)).get();
+                try (var consumer = new KafkaConsumer<String, String>(
+                        Map.of(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, proxyAddress, ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class,
+                                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class, ConsumerConfig.GROUP_ID_CONFIG, "my-group-id",
+                                ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"))) {
+                    consumer.subscribe(Set.of(TOPIC_2));
+                    var records = consumer.poll(Duration.ofSeconds(10));
+                    assertEquals(1, records.count());
+                    assertEquals(SYNTACTICALLY_INCORRECT_JSON, records.iterator().next().value());
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testPartiallyInvalidJsonTransactionalAllRejected(KafkaCluster cluster, Admin admin) throws Exception {
+        String proxyAddress = "localhost:9192";
+
+        admin.createTopics(List.of(new NewTopic(TOPIC_1, 1, (short) 1), new NewTopic(TOPIC_2, 1, (short) 1))).all().get();
+
+        var config = baseConfigBuilder(proxyAddress, cluster.getBootstrapServers()).addNewFilter().withType("ProduceValidator").withConfig(
+                Map.of("forwardPartialRequests", true, "rules", List.of(Map.of("topicNames", List.of(TOPIC_1, TOPIC_2), "valueRule",
+                        Map.of("allowsNulls", true, "syntacticallyCorrectJson", Map.of("validateObjectKeysUnique", true))))))
+                .endFilter()
+                .build().toYaml();
+
+        try (var ignored = startProxy(config)) {
+            try (var producer = getProducer(proxyAddress,
+                    Map.of(ProducerConfig.LINGER_MS_CONFIG, 100, ProducerConfig.TRANSACTIONAL_ID_CONFIG, UUID.randomUUID().toString()))) {
```

Comment:
Thinking about test robustness.
* Why choose such a short linger? Why not explicitly set the batch size to something bigger than the records you are about to send?
* The test assumes that the topic partitions are on the same broker.  Adding an assertion that checks the cluster has exactly one broker would make this explict.

Reply:
Done, I've switched to a higher linger.ms with explicit flushing. This should be more reliable on slow hardware, thanks. The batch is 16K by default but I've made it explicit since it's relevant to the tests. Also added in the broker size assertion.